src/components/MAPI/types.ts

```diff
@@ -19,6 +19,6 @@ export type NodePoolList =
   | capiv1alpha3.IMachineDeploymentList
   | capiexpv1alpha3.IMachinePoolList;
 
-export type ProviderNodePool = capzexpv1alpha3.IAzureMachinePool;
+export type ProviderNodePool = capzexpv1alpha3.IAzureMachinePool | undefined;
```

Comment:
Provider-specific node pools can now be optional. When a machine pool is deleting, its provider-specific resource will be deleted first.

In order to display machine pools that are currently deleting, we need to take into account that the provider-specific resources have already been deleted.